src/care/LoopFuser.h

```diff
@@ -1254,13 +1254,23 @@ void LoopFuser<REGISTER_COUNT, XARGS...>::registerAction(const char * fileName,
 
 
 
-#define FUSIBLE_KERNEL_R_END return 0;}); }
+#define FUSIBLE_KERNEL_R_END return 0;}); } FUSIBLE_PHASE_RESET
 #define FUSIBLE_KERNEL_PHASE_R_END FUSIBLE_KERNEL_R_END
 #define FUSIBLE_KERNEL_END FUSIBLE_KERNEL_R_END
 
 #define FUSIBLE_KERNEL_PHASE(PRIORITY) FUSIBLE_KERNEL_PHASE_R(PRIORITY, CARE_DEFAULT_LOOP_FUSER_REGISTER_COUNT)
 
-#define FUSIBLE_PHASE_RESET FusedActionsObserver::getActiveObserver()->reset_phases(__FILE__, __LINE__);
+#define FUSIBLE_PHASE_RESET   if (FusedActions::flush_now) { \
```

Comment:
Could we make this a function rather than a macro?

Reply:
Would that be a static method in FusedActions?  Or FusedActionsObserver?   It would have to be defined later, as it needs to get instances of LoopFuser.